pylint/config/__init__.py

```diff
@@ -56,14 +58,14 @@
 ]
 
 USER_HOME = os.path.expanduser("~")
-if "PYLINTHOME" in os.environ:
-    PYLINT_HOME = os.environ["PYLINTHOME"]
+PYLINT_HOME = os.getenv("PYLINTHOME")
+if PYLINT_HOME:
     if USER_HOME == "~":
         USER_HOME = os.path.dirname(PYLINT_HOME)
 elif USER_HOME == "~":
     PYLINT_HOME = ".pylint.d"
 else:
-    PYLINT_HOME = os.path.join(USER_HOME, ".pylint.d")
```

Comment:
Maybe we could check if the old default directory exists and warn the user that it can now be deleted ? Maybe even copy the old cache to the new directory?

Reply:
How about mentioning it in ChangeLog or faq.rst?

Reply:
That would be nice too :) But I think 90 to 95% of users do not read the changelog and need something in pylint to tell them.